docs(js): Create Quick Start guide for Nuxt manual setup (#13324)
getsentry/sentry-docs: docs/platforms/javascript/guides/nuxt/manual-setup.mdx

  replaysSessionSampleRate: 0.1,
  replaysOnErrorSampleRate: 1.0,
});
```

We recommend you store your Sentry [Data Source Name](/concepts/key-terms/dsn-explainer/) (DSN) in an environment variable and configure it via the Nuxt runtime config like so:

```javascript {filename:nuxt.config.ts}
export default defineNuxtConfig({
  modules: ["@sentry/nuxt"],
  runtimeConfig: {
    public: {
      sentry: {
        dsn: process.env.SENTRY_DSN_PUBLIC, // Use a public environment variable for the DSN
      },
    },
  },
});
```

This allows you to access the DSN using `useRuntimeConfig().public.sentry.dsn`.

### Configure Server-side Sentry

Add a `sentry.server.config.ts` file to the root of your project and add the following initialization code to it:

```javascript {filename:sentry.server.config.ts}
import * as Sentry from "@sentry/nuxt";

Sentry.init({
  dsn: "___PUBLIC_DSN___",

  // Set tracesSampleRate to 1.0 to capture 100%
  // of transactions for tracing.
  // We recommend adjusting this value in production
  // Learn more at
  // https://docs.sentry.io/platforms/javascript/configuration/options/#traces-sample-rate

>   tracesSampleRate: 1.0,

Reviewer: This needs to be adapted to the onboarding options, currently it is always visible

});
```

We recommend you store your Sentry [Data Source Name](/concepts/key-terms/dsn-explainer/) (DSN) in an environment variable.

<Expandable title="How to access environment variables in sentry.server.config.ts">

> Since Sentry on the server side needs to be loaded before `useRuntimeConfig()` is fully available, environment variables are not directly accessible via `process.env`. To make sure your environment variables are available, use one of these methods:

Reviewer 2: The old docs are not very clear on this, but `useRuntimeConfig` and `process.env` are two different things. `useRuntimeConfig` is something Nuxt provides and as this is not available, people have to use `process.env`.

```suggestion
Since Sentry on the server side needs to be loaded before `useRuntimeConfig()` is fully available, environment variables are only accessible via `process.env`. To make sure your environment variables are available, use one of these methods:
```

or use the `dotenv` package:

```javascript {tabTitle: Server Config} {filename:sentry.server.config.ts} {1,3}
import dotenv from "dotenv";

dotenv.config();

// ... rest of the file
```

</Expandable>

**Sentry's server-side monitoring doesn't work in development mode**. To enable it, you first need to build your application and then load the Sentry server-side config using the `--import` flag when running your application:

```
# Start your app after building your project with `nuxi build`
node --import ./.output/server/sentry.server.config.mjs .output/server/index.mjs
```

Check out the <PlatformLink to="/install/cli-import">`--import` CLI flag</PlatformLink> docs for setup instructions.

## Step 3: Add Readable Stack Traces With Source Maps (Optional)

> The Sentry Nuxt Module uses the [Sentry Vite Plugin](https://www.npmjs.com/package/@sentry/vite-plugin) to upload source maps for both server and client builds.

Reviewer 2: This is a bit outdated :D

```suggestion
The Sentry Nuxt Module uses the [Sentry Vite Plugin](https://www.npmjs.com/package/@sentry/vite-plugin) and the [Sentry Rollup Plugin](https://www.npmjs.com/package/@sentry/rollup-plugin) to upload source maps for both server and client builds.
```

or use the `dotenv` package:

```javascript {tabTitle: Server Config} {filename:sentry.server.config.ts} {1,3}
import dotenv from "dotenv";

dotenv.config();

// ... rest of the file
```

</Expandable>

**Sentry's server-side monitoring doesn't work in development mode**. To enable it, you first need to build your application and then load the Sentry server-side config using the `--import` flag when running your application:

```
# Start your app after building your project with `nuxi build`
node --import ./.output/server/sentry.server.config.mjs .output/server/index.mjs
```

Check out the <PlatformLink to="/install/cli-import">`--import` CLI flag</PlatformLink> docs for setup instructions.

<Alert level="warning">
  In the beta state of the Nuxt SDK, some features may not work with certain
  deployment providers. Check the progress on GitHub: [Compatibility with
  different Deployment

>   Platforms](https://github.com/getsentry/sentry-javascript/issues/14029)

Author: The linked issue is closed -> is this still relevant @s1gr1d ?

Reviewer 2: No, this can be deleted. It's a general limitation with ESM and people can opt in to [Limited Server Tracing](https://docs.sentry.io/platforms/javascript/guides/nuxt/install/limited-server-tracing/).

<OnboardingOption optionId="performance">
### Tracing

To test tracing, create a test API route `server/api/sentry-example.get.ts`:

```js {tabTitle:Nitro} {filename:server/api/sentry-example.get.ts}
export default defineEventHandler((event) => {
  throw new Error("Sentry Example API Route Error");
});
```

Then update the test page by including a new button that executes a function to fetch your API route:

```html {tabTitle:Vue} {filename:pages/example-error.vue}
<script setup>
  import * as Sentry from "@sentry/nuxt";

  function triggerClientError() {
    throw new Error("Nuxt Button Error");
  }

  function getSentryData() {
    Sentry.startSpan(
      {
        name: "Example Frontend Span",
        op: "test",
      },
      async () => {
        await $fetch("/api/sentry-example");
      }
    );
  }
</script>

<template>
  <button id="errorBtn" @click="triggerClientError">Throw Client Error</button>
  <button type="button" @click="getSentryData">Throw Server Error</button>
</template>
```

Once you have your test code in place, you need to build your project since Sentry's server-side monitoring doesn't work in development mode.

> Then start your app and make sure to load Sentry on the server side by explicitly adding it via `--import`.

Reviewer 2: We could link to those docs here: https://docs.sentry.io/platforms/javascript/guides/nuxt/install/cli-import/
```suggestion
Then start your app and make sure to load Sentry on the server side by explicitly adding the Sentry server config in the build output via `--import`.
```

After running your project:

> 1. Open your test page in a browser (for most Nuxt applications, this will be at localhost)

Reviewer 2: I am not a fan of just writing `localhost` as this is the same for every local application. But we could add the port here, as this is something that differers from setup to setup. E.g. Svelte uses [port 5173](https://svelte.dev/docs/kit/creating-a-project) as default and Nuxt [port 3000](https://nuxt.com/docs/api/nuxt-config#port).
```suggestion
1. Open your test page in a browser (for most Nuxt applications, this will be at localhost:3000)
```

2. Click the "Throw Client Error" button to trigger an error in the frontend
3. Click the "Throw Server Error" button to trigger an error within the API route and start a performance trace to measure the time it takes for the API request to complete.

</OnboardingOption>

### View Captured Data in Sentry

Now, head over to your project on [Sentry.io](https://sentry.io) to view the collected data (it takes a couple of moments for the data to appear).

<PlatformContent includePath="getting-started-verify-locate-data" />

## Next Steps

> At this point, you should have integrated Sentry into your Next.js application and should already be sending data to your Sentry project.

Reviewer 2: ```suggestion
At this point, you should have integrated Sentry into your Nuxt application and should already be sending data to your Sentry project.
```

Now's a good time to customize your setup and look into more advanced topics. Our next recommended steps for you are:

- Learn how to [manually capture errors](/platforms/javascript/guides/nuxt/usage/)
- Continue to [customize your configuration](/platforms/javascript/guides/nuxt/configuration/)
- Get familiar with [Sentry's product features](/product) like tracing, insights, and alerts
- Learn how to [track your Vue components or your Pinia store](/platforms/javascript/guides/nuxt/features/)

<Expandable permalink={false} title="Are you having problems setting up the SDK?">

> 

Reviewer 2: I would still add a link to the installations methods here - this is something that a lot of people miss. However, it's not really a problem with "setting up the SDK", but rather a problem with "not having Sentry initialized on the server when deployed".

Author: 👍I added the link to the installation methods page to this list -- they still fit in there topic-wise
